poe_set_port.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/PuerkitoBio/goquery"
+	"io"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type PoeSetPowerCommand struct {
+	Address   string `required:"" help:"the Netgear switch's IP address or host name to connect to" short:"a"`
+	Ports     []int  `required:"" help:"switch port(s) to set power for" short:"p" name:"port"`
+	PortPwr   string `optional:"" help:"power state for port (ex: enable, disable)" short:"s" name:"power"`
+	PwrMode   string `optional:"" help:"port power mode (ex: 802.3af, 802.3at)" name:"mode"`
+	PortPrio  string `optional:"" help:"port priority (ex: low, high, critical)" name:"priority"`
+	LimitType string `optional:"" help:"port power limit type (ex: none, class, user)" name:"limit-type"`
+	PwrLimit  string `optional:"" help:"port power limit (W)" name:"pwr-limit"`
+	DetecType string `optional:"" help:"port detection type (ex: IEEE 802, legacy)" name:"detect-type"`
+}
+
+type PoeExt struct {
+	Hash         string
+	PortMaxPower string
+}
+
+func (poe *PoeSetPowerCommand) Run(args *GlobalOptions) error {
+
+	poeExt := &PoeExt{}
+	var adminMode string
+
+	settings, err := requestPoeConfiguration(args, poe.Address, poeExt)
+	if err != nil {
+		return err
+	}
+
+	for _, switchPort := range poe.Ports {
+		if switchPort > len(settings) || switchPort < 1 {
+			return errors.New("port out of range.")
```

Comment:
This one is still missing ... but I will add it later.

```return errors.New(fmt.Sprintf("given port id %d, doesn't fit in range 1..%d", switchPort, len(settings)))```

Reply:
This must have been dropped during a rebase of some minor WIP commits, apologies and thanks for catching it 😅